File: c++ code/week1/examples/binary image processing/binary image processing/erosion.cpp
#include <opencv2/opencv.hpp>

int main3() {
	cv::Mat image = cv::imread("../../../../../images/truth.png", cv::IMREAD_COLOR);

	int erosionSize = 6;
	cv::Mat element = cv::getStructuringElement(cv::MORPH_CROSS,
		cv::Size(2 * erosionSize + 1, 2 * erosionSize + 1),
		cv::Point(erosionSize, erosionSize));

	cv::Mat erodedImage;
	cv::erode(image, erodedImage, element);

	cv::imshow("Original Image", image);
	cv::imshow("Eroded Image", erodedImage);

	cv::waitKey(0);
	return EXIT_SUCCESS;
}

File: python code/binary image processing/erosion.py
import cv2

# read image
image = cv2.imread("../../images/truth.png", cv2.IMREAD_COLOR)

# get structuring kernel which used for erosion
erosionSize = 6
element = cv2.getStructuringElement(shape=cv2.MORPH_CROSS,
                                    ksize=(2 * erosionSize + 1, 2 * erosionSize + 1),
                                    anchor=(erosionSize, erosionSize))

# erode the image
erodedImage = cv2.erode(src=image, kernel=element)

# display image
cv2.imshow("Original Image", image)
cv2.imshow("Eroded Image", erodedImage)

cv2.waitKey(0)
